app/models/katello/activation_key.rb

```diff
@@ -99,11 +99,32 @@ def products
           Rails.logger.error("Pool #{pool.id} is missing its subscription id.")
         end
       end
-      all_products.flatten!
+      all_products.uniq.flatten
+    end
+
+    def all_products
+      Katello::Product.joins(:subscriptions => :pools).where(:organization_id => organization.id).enabled.uniq
     end
 
     def available_content
-      self.products ? self.products.map(&:available_content).flatten.uniq { |product| product.content.id } : []
+      self.products.map(&:available_content).flatten.uniq { |product| product.content.id }
+    end
+
+    def available_product_content(content_access_mode_all = false, content_access_mode_env = false)
+      if content_access_mode_env && environment_id && content_view_id
+        version = ContentViewVersion.in_environment(environment_id).where(:content_view_id => content_view_id).first
+        content_view_version_id = version.id
```

Comment:
You can't set a CV/LE on a key if its not published to that env, however always good to handle that case !